sdk/servicebus/service-bus/src/core/batchingReceiver.ts

```diff
@@ -54,6 +41,42 @@ export class BatchingReceiver extends MessageReceiver {
    */
   constructor(context: ClientEntityContext, options?: ReceiveOptions) {
     super(context, ReceiverType.batching, options);
+
+    this._batchingReceiverLite = new BatchingReceiverLite(
+      context,
+      async (abortSignal?: AbortSignalLike): Promise<MinimalReceiver> => {
+        let lastError: Error | AmqpError | undefined;
+
+        const rcvrOptions = this._createReceiverOptions(false, {
+          onError: (context) => {
+            lastError = context.error || context?.session?.error || context?.receiver?.error;
+          },
+          onSessionError: (context) => {
+            lastError = context.error || context?.session?.error || context?.receiver?.error;
+          },
+          // ignored for now - the next call will just fail so they'll get an appropriate error from somewhere else.
+          onClose: async () => {},
+          onSessionClose: async () => {},
+          // we don't add credits initially so we don't need to worry about handling any messages.
+          onMessage: async () => {}
+        });
+
+        await this._init(rcvrOptions, abortSignal);
+
+        if (lastError != null) {
+          throw lastError;
```

Comment:
What kind of errors do we expect to track using this `lastError`?
Once credits are added, we have separate onError handlers.
So, this is for after we create the link, but before we add the credits?

Reply:
Yes, that's exactly it. A _really_ small window of time but not covering it would lead to potential cover up later so I figured it's simpler.

One spot I did see it was when I misconfigured the initial creation of the receiver. So some of these "failure to launch" type things will show up here as well.

Reply:
Can you share more about this misconfiguration?

Reply:
Sure.

When I was refactoring I took over a bit more of the initialization procedure, which meant I was actually crafting my own receiver options (down to the flags that control autoaccept, etc..). It's not something that would normally happen but having error handlers to catch these kinds of issues seemed like a good idea.